src/playground/sample_data_ingestion/sample_data_plugins/sample_data_plugin.py

```diff
@@ -0,0 +1,145 @@
+        if self.base_url.endswith("/"):
+          self.base_url = self.base_url[:-1]
+        self.url = self.base_url + ":9200/"
+    
+    # If payload is a zipped config file, unzip it and return unzipped filename
+    def unzip(self):
+        if type(self.payload) is not str or (type(self.payload) is str and ".json" not in self.payload):
+            print("Payload is not a filename or is not a .json file")
+            return None
+        if type(self.payload) is str and ".gz" not in self.payload:
+            print("Payload is already unzipped")
+            return self.payload
+        with gzip.open(self.payload, 'rt') as fin:
+            with open(self.payload.split(".gz")[0], 'wb') as fout:
+                copyfileobj(fin, fout)
+        return self.payload.split(".gz")[0]
+
+    # Utility function to convert payload to a dict
+    def convert_payload(self, payload, plugin_config_directory_name):
+        if type(payload) is str and ".json" not in payload:
+            payload = loads(payload)
+
+        # If payload is a config file, navigate to the config file and load contents to payload (unzip if necessary)
+        if type(self.payload) is str and ".json" in self.payload:
+            dir_path = path.dirname(path.realpath(__file__))
+            dir_path = dir_path.split("/sample_data_plugin")[0]
+            config_file = self.unzip()
+            dir_path += "/config/" + plugin_config_directory_name + "/" + config_file
+            with open (dir_path, "r") as f:
+                payload = load(f)
+
+            # If the file was previously unzipped, remove the unzipped file
+            if config_file != self.payload:
+                remove(config_file)
+
+        return payload
+
+
+"""
+AnomalyDetection class: Inherited class from Plugin class
+Arguments (only for AnomalyDetection):
+    - target_index: the index to store anomaly detector results
+    - days_ago: how far back the historical analysis will start (default = 7 days)
+"""
+class AnomalyDetection(Plugin):
+    def __init__(self, index_name, payload, base_url, target_index = None, days_ago = 7):
+        super().__init__(index_name, payload, base_url)
+        self.url += "_plugins/_anomaly_detection/"
+        self.target_index = target_index
+        self.days_ago = days_ago
+    
+    # Creates detector using the initial arguments
+    def create_detector(self):
+        payload = self.convert_payload(self.payload, "ad_plugin")
+        endpoint = self.url + "detectors"
+        payload["indices"] = self.index_name
+        
+        # Convert indices to an array type
+        if type(self.index_name) is str:
+            payload["indices"] = [self.index_name]
+        
+        # Modify payload and send API call
+        if self.target_index:
+            payload["result_index"] = self.target_index
+        payload = dumps(payload)
+        response = request("POST", endpoint, headers = self.headers, data = payload, verify = False)
```

Comment:
if request failed, do we catch the exception for the request ?

Reply:
Done, error catching is impleemented for each request